scio-core/src/main/scala/com/spotify/scio/values/ErrorSink.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.scio.values
+
+import com.spotify.scio.ScioContext
+import org.apache.beam.sdk.transforms.PTransform
+import org.apache.beam.sdk.transforms.errorhandling.{BadRecord, ErrorHandler}
+import org.apache.beam.sdk.values.{PCollection, PCollectionTuple, TupleTag}
+
+/**
+ * A sink for error records.
+ *
+ * Once the [[sink]] is materialized, the [[handler]] must not be used anymore.
+ */
+sealed trait ErrorSink {
+  def handler: ErrorHandler[BadRecord, _]
```

Comment:
could `def handler` be `private[scio]`? not sure when a user would need to access this

Reply:
This is the API exposed by beam. As mentioned in the description we do not pass the `ErrorSink` directly.
```scala
sc.bigQueryStorageFormat[MyType](
  table,
  format,
  errorHandler = errorSink.handler
)
```

I was thinking of adding to the `ScioContext` a beam java like API too

```scala
def registerBadRecordErrorHandler[T](handler: PTransform[PCollection[BadRecord], T] sinkTransform): BadRecordErrorHandler[OutputT]
```